### PR TITLE
Update Docker to cudnn6 and fix src mapping

### LINKED ISSUE #6750: Docker keras image:  ImportError: No module named 'keras'
Hi -

   I recently installed the keras docker image that is served from https://github.com/fchollet/keras/tree/master/docker.

using the makefile, when I invoke it with:
make notebook GPU=0

the notebook comes up fine.  However, when I invoke a cell that has:

import keras

I get this error:

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-c74e2bd4ca71> in <module>()
----> 1 import keras

ImportError: No module named 'keras'

### PR DESCRIPTION
**Pull request by jussihuotari**
I had to make a few tweaks to the Keras Docker image to get it up and running. 

* Upgraded cuDNN5 to cuDNN6 following [Tensorflow's example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/docker/Dockerfile.gpu#L1). 
* Changed the workspace directory to be able to `import keras`. This fixes #6750.
* Added `SRC` as a configurable parameter in the Makefile to be able to use something like `make notebook GPU=0 BACKEND=tensorflow DATA=~/project/foo/data SRC=~/project/foo`

Before these tweaks I got `libcudnn.so.6: cannot open shared object file: No such file or directory`

After the edits, I was able to run `make test` with the following results:

```
E ValueError: Unexpectedly found an instance of type `<class 'theano.gpuarray.type.GpuArraySharedVariable'>`. Expected a symbolic tensor instance.

keras/backend/theano_backend.py:211: ValueError

1 failed, 505 passed, 1 skipped, 212 warnings
```

I believe Theano works with cuDNN6 and that it is officially supported: https://github.com/Theano/Theano/blob/master/theano/gpuarray/cudnn_defs.py#L13

In other words: I don't know whether the test failure is caused by cuDNN version change or not, but I think not.

### CODE REVIEW

**Comment by fchollet** [comment]
Please update following https://github.com/fchollet/keras/pull/7884/files

**Comment by jussihuotari** [comment]
Good catch. Updated.
